analyses/create-subset-files/01-create_subset_files.Rmd

```diff
@@ -0,0 +1,370 @@
+controlfreec <- readr::read_tsv(file.path(data_dir, controlfreec_file))
+```
+
+Extract the identifiers from the SEG files.
+
+```{r}
+sample_id_list$cnvkit <- unique(cnvkit$ID)
+sample_id_list$controlfreec <- unique(controlfreec$ID)
+```
+
+## Fusion
+
+```{r}
+arriba <- readr::read_tsv(file.path(data_dir, arriba_file))
+starfusion <- readr::read_tsv(file.path(data_dir, starfusion_file))
+```
+
+```{r}
+sample_id_list$arriba <- unique(arriba$tumor_id)
+sample_id_list$starfusion <- unique(starfusion$tumor_id)
+```
+
+## Gene expression
+
+```{r}
+kallisto <- readr::read_rds(file.path(data_dir, kallisto_file))
+rsem <- readr::read_rds(file.path(data_dir, rsem_file))
+```
+
+```{r}
+# the first two columns are transcript and gene identifiers, respectively
+sample_id_list$kallisto <- colnames(kallisto)[3:ncol(kallisto)]
+# the first column contains gene identifiers
+sample_id_list$rsem <- colnames(rsem)[2:ncol(kallisto)]
+```
+
+## SNV
+
+SNV files are in [annotated MAF format](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/doc/format/vep-maf.md).
+
+These are a bit unwieldy to read in, so we'll split up the chunks below.
+
+```{r}
+mutect2 <- data.table::fread(file.path(data_dir, mutect2_file),
+                             skip = 1,  # skip version string
+                             data.table = FALSE)
+```
+
+```{r}
+strelka2 <- data.table::fread(file.path(data_dir, strelka2_file),
+                              skip = 1,  # skip version string
+                              data.table = FALSE)
+```
+
+```{r}
+sample_id_list$mutect2 <- unique(mutect2$Tumor_Sample_Barcode)
+sample_id_list$strelka2 <- unique(strelka2$Tumor_Sample_Barcode)
+```
+
+## SV 
+
+The Manta SV file is in [annotated Manta TSV format](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/doc/format/manta-tsv-header.md).
+
+```{r}
+manta <- readr::read_tsv(file.path(data_dir, manta_file))
+```
+
+```{r}
+sample_id_list$manta <- unique(manta$Kids.First.Biospecimen.ID.Tumor)
+```
+
+```{r}
+#TODO: LUMPY SV
+```
+
+## Select samples and subset files
+
+### Map to participant identifiers
+
+These sample identifiers are all biospecimen identifiers.
+The tools that use different experiment strategies will have no overlapping
+biospecimen identifiers.
+Thus, we'll need to map _biospecimen_ identifiers to _participant_ identifiers.
+We'll use the histologies file to do this.
+
+```{r}
+histologies_df <- readr::read_tsv(file.path(data_dir, histologies_file))
+```
+
+```{r}
+participant_id_list <- lapply(
+  sample_id_list,
+  function(sample_ids) {
+    histologies_df %>%
+      dplyr::filter(Kids_First_Biospecimen_ID %in% sample_ids) %>% 
+      dplyr::pull(Kids_First_Participant_ID)
+  }
+)
+```
+
+Now we can take the intersection of elements of `participant_id_list` to
+identify participant identifiers represented in each of the files we're trying
+to subset.
+
+```{r}
+participants_in_all <- Reduce(intersect, participant_id_list)
```

Comment:
To prevent people from shooting themselves in the foot, and since there are a lot of functions named "Reduce", you could specify that this is the Reduce from `base::Reduce`. I thought maybe you were using the `BiocGenerics::Reduce` for a sec. People *should* be refreshing their R sessions, but we know this often won't happen.